src/openeo_odc/map_processes_odc.py

```diff
@@ -48,7 +48,7 @@ def map_load_collection(id, process):
                     params['y'] = (lowLat,highLat)
     if 'temporal_extent' in process['arguments']:
         def exclusive_date(date):
-            return str(np.datetime64(date) - np.timedelta64(1, 'D')).split(' ')[0] # Substracts one day
+            return str(np.datetime64(date) - np.timedelta64(1, 'D')) + 'Z' # Substracts one day
```

Comment:
Why did you add the `Z`? This looks quite weird e.g. `'time': ['2018-06-15', '2018-06-16Z']` because the start time is not followed by a `Z`. And I'm not completely sure if we need it..?